File: backend/LibraryManagement.Core/Book.cs
namespace LibraryManagement.Core;

/// <summary>
/// Represents a book in the library.
/// </summary>
public class Book
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Author { get; set; } = string.Empty;
    public string? Description { get; set; }
}

File: frontend/src/types/book.ts
/** Book as returned by the API */
export interface Book {
  id: number
  title: string
  author: string
  description: string | null
}

/** Input for creating a new book */
export interface CreateBookInput {
  title: string
  author: string
  description?: string | null
}

/** Input for updating an existing book */
export interface UpdateBookInput {
  title: string
  author: string
  description?: string | null
}

/** API error response shape */
export interface ApiError {
  error?: string
  title?: string
  errors?: Record<string, string[]>
}
